### PR TITLE
chore: upgrade taskgraph

### PR DESCRIPTION
**Pull request by matt-boris**
Upgrades taskgraph from v1.2.0 --> v3.4.0.

### CODE REVIEW

File: .taskcluster.yml [hunk 1]
```diff
@@ -42,11 +42,40 @@ tasks:
                       $if: 'tasks_for == "github-release"'
                       then: '${event.release.target_commitish}'
 
+          base_ref:
```
Comment (lotas) [inline .taskcluster.yml:45]: this one made the staging release fail

I had  to add an override down below to point it to the `refs/head/main` on staging release, as `run-task` was failing to checkout given reference. It didn't exist:

```
[task 2022-11-03T10:19:35.498Z] 2022-11-03 10:19:35,497 - INFO - Using base_ref "refs/heads/staging-release/v44.23.1" to determine latest common revision...
[task 2022-11-03T10:19:35.524Z] fatal: Not a valid object name refs/heads/staging-release/v44.23.1
```

after switching it to the `main` ref it was working fine:

```
[vcs 2022-11-03T11:59:45.039Z] executing ['git', 'fetch', 'origin', 'refs/heads/main']
[vcs 2022-11-03T11:59:47.736Z] From https://github.com/taskcluster/staging-releases
[vcs 2022-11-03T11:59:47.736Z]  * branch                main       -> FETCH_HEAD
[vcs 2022-11-03T11:59:47.782Z] executing ['git', 'checkout', 'refs/heads/main']
[vcs 2022-11-03T11:59:47.842Z] HEAD is now at 9428ec6d4 README
[vcs 2022-11-03T11:59:47.845Z] executing ['git', 'fetch', '--no-tags', 'https://github.com/taskcluster/staging-releases', 'refs/heads/staging-release/v44.23.1']
[vcs 2022-11-03T11:59:52.384Z] From https://github.com/taskcluster/staging-releases
[vcs 2022-11-03T11:59:52.385Z]  * branch                staging-release/v44.23.1 -> FETCH_HEAD
[vcs 2022-11-03T11:59:52.427Z] executing ['git', 'checkout', '-f', '-B', 'refs/heads/staging-release/v44.23.1', '948aacbb81c15907c796bddf43781072b123c443']
[vcs 2022-11-03T11:59:53.070Z] Previous HEAD position was 9428ec6d4 README
[vcs 2022-11-03T11:59:53.082Z] Switched to a new branch 'refs/heads/staging-release/v44.23.1'
[vcs 2022-11-03T11:59:53.090Z] cleaning git checkout...
[vcs 2022-11-03T11:59:53.091Z] executing ['git', 'clean', '-nxdff']
[vcs 2022-11-03T11:59:53.144Z] removing []
[vcs 2022-11-03T11:59:53.145Z] successfully cleaned git checkout!
```

I'm not fully sure why it couldn't find that ref originally:
`fatal: Not a valid object name refs/heads/staging-release/v44.23.1`

```
root@c7e1e0c28187:~/checkouts/src# git show-ref
9428ec6d400a57340fd801585c4f1e37fd33bae4 refs/heads/main
9428ec6d400a57340fd801585c4f1e37fd33bae4 refs/remotes/origin/HEAD
9428ec6d400a57340fd801585c4f1e37fd33bae4 refs/remotes/origin/main
a0904a165724647a5e65e2abf2d703309fab55ad refs/remotes/origin/staging-release/v1.2.3
f7e4d8cf0f48291ae05385e368f952b3a6ac73d5 refs/remotes/origin/staging-release/v31.0.1
f31593423744524d123526325d9d3552bc9e2b64 refs/remotes/origin/staging-release/v32.0.1
98ecfe0a7690e724ceb9c024a9132a60bac6374e refs/remotes/origin/staging-release/v34.0.1
cd5b3740c8e78d962a0c85517672e68196bcd3a6 refs/remotes/origin/staging-release/v35.0.0
6bac06cd817ae2f7cc1d0fa7269c19386d82bd4f refs/remotes/origin/staging-release/v36.0.0
3a0394378f38397c9d20336c1eb47bd786e14454 refs/remotes/origin/staging-release/v38.0.4
42c036ead69fa01f8ffb0ba8d27f4b2ac6e7ab84 refs/remotes/origin/staging-release/v39.0.0
6c001303e4fd62b8b3a85471bac61110feee3c5a refs/remotes/origin/staging-release/v41.0.0
f3399351d72043d5d0cc8486f52d7f6c4ad40516 refs/remotes/origin/staging-release/v43.2.0
447e9ad3918081c54df0188f29692b8139dfed2c refs/remotes/origin/staging-release/v44.1.0
b239801344df21f733e3361991a072f533e23cc4 refs/remotes/origin/staging-release/v44.1.1
21ed47dc8560486147a8d9a0c5271e62bf1f3f8d refs/remotes/origin/staging-release/v44.12.1
102eb06459874457f0733ab20635b63d9d6d597e refs/remotes/origin/staging-release/v44.13.1
d04f172cf3a4185a6df73729fbd8ee743f28a125 refs/remotes/origin/staging-release/v44.14.1
4acee0e00d53c8b55e160dcc1b0d4174ffe446e1 refs/remotes/origin/staging-release/v44.19.1
e559418c7f50eeaacc950c25b48c7048cb98d0bc refs/remotes/origin/staging-release/v44.2.0
a709c5ff24e61e5cacc25157dae53774535a8b64 refs/remotes/origin/staging-release/v44.20.0
55f8f39198d11f6fcaf2d5ac2693d57858afb677 refs/remotes/origin/staging-release/v44.20.2
08124401ec89eccb40572e61c78655bd8795670a refs/remotes/origin/staging-release/v44.20.4
96b2dd369529a9282d2265e88f99fdf3331139a1 refs/remotes/origin/staging-release/v44.21.0
948aacbb81c15907c796bddf43781072b123c443 refs/remotes/origin/staging-release/v44.23.1
a41317d35ea198d81e5853e7125fce43c42f74a9 refs/remotes/origin/staging-release/v44.6.0
```

maybe `run-task` doesn't checkout all refs, but git should be aware of those.. idk

File: .taskcluster.yml [hunk 2]
```diff
@@ -15,25 +15,25 @@ tasks:
               $if: 'tasks_for in ["github-push", "github-release"]'
               then: '${event.repository.html_url}'
               else:
-                  $if: 'tasks_for == "github-pull-request"'
+                  $if: 'tasks_for[:19] == "github-pull-request"'
```
Comment (lotas) [inline .taskcluster.yml:18]: do you remember why we need to do the string slice here? I don't think github service will send any strange values for the `tasks_for`

Comment (matt-boris) [reply .taskcluster.yml:18]: I was just doing similar to taskgraph https://github.com/taskcluster/taskgraph/blob/main/.taskcluster.yml

Comment (lotas) [reply .taskcluster.yml:18]: aah. this is probably because of the new value that Ahal added - `github-pull-request-unrestricted` .. ok

Comment (matt-boris) [reply .taskcluster.yml:18]: Yeah, I think you're right. That makes more sense

**Review by lotas — APPROVED** [review]

**Pull request merged**